### ISSUE TITLE
Cannot log in with boundary desktop

### ISSUE THREAD

**Issue by Abfelbaum** [hashicorp/boundary-ui#1705]
**Describe the bug**

When I try to log in to my boundary instance with boundary desktop the following error occures:

![image](https://github.com/hashicorp/boundary-ui/assets/26802234/0c897d3a-d682-48ab-81f7-00e668b8446e)

**To Reproduce**
1. Open Boundary Desktop
2. Connect to https://boundary.abfelbaum.dev
3. Log In

**Expected behavior**
Boundary desktop should log in to my account and I should be able to connect to my targets

**Additional context**
I ran boundary desktop with the `--enable-logging` option and got the following logs:

```
[7428:0605/130025.433:INFO:CONSOLE(3697)] "Refused to connect to 'https://boundary.abfelbaum.dev/v1/scopes' because it violates the following Content Security Policy directive: "connect-src 'self'".
", source: serve://boundary/assets/vendor-c45cd3fe4fc053d713149ecaffed61a4.js (3697)
[7428:0605/130034.470:INFO:CONSOLE(0)] "Access to XMLHttpRequest at 'https://boundary.abfelbaum.dev/v1/scopes' from origin 'serve://boundary' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource.", source: serve://boundary/#/scopes/global/authenticate/ampw_pducx1eWW7 (0)
[7428:0605/130034.471:INFO:CONSOLE(3697)] "Uncaught (in promise) TypeError: Network request failed", source: serve://boundary/assets/vendor-c45cd3fe4fc053d713149ecaffed61a4.js (3697)
[7428:0605/130034.474:INFO:CONSOLE(0)] "Access to XMLHttpRequest at 'https://boundary.abfelbaum.dev/v1/scopes/global' from origin 'serve://boundary' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource.", source: serve://boundary/#/scopes/global/authenticate/ampw_pducx1eWW7 (0)
[7428:0605/130034.474:INFO:CONSOLE(0)] "Access to XMLHttpRequest at 'https://boundary.abfelbaum.dev/v1/scopes' from origin 'serve://boundary' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource.", source: serve://boundary/#/scopes/global/authenticate/ampw_pducx1eWW7 (0)
[7428:0605/130034.474:INFO:CONSOLE(0)] "Access to XMLHttpRequest at 'https://boundary.abfelbaum.dev/v1/scopes?scope_id=global' from origin 'serve://boundary' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource.", source: serve://boundary/#/scopes/global/authenticate/ampw_pducx1eWW7 (0)
[7428:0605/130034.474:INFO:CONSOLE(4799)] "Uncaught (in promise) Error: Attempted to handle event `notFound` on <scope:global> while in state root.loaded.created.uncommitted. ", source: serve://boundary/assets/vendor-c45cd3fe4fc053d713149ecaffed61a4.js (4799)
[7428:0605/130034.474:INFO:CONSOLE(0)] "Access to XMLHttpRequest at 'https://boundary.abfelbaum.dev/v1/scopes/global' from origin 'serve://boundary' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource.", source: serve://boundary/#/scopes/global/authenticate/ampw_pducx1eWW7 (0)
[7428:0605/130034.474:INFO:CONSOLE(0)] "Access to XMLHttpRequest at 'https://boundary.abfelbaum.dev/v1/scopes?scope_id=global' from origin 'serve://boundary' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource.", source: serve://boundary/#/scopes/global/authenticate/ampw_pducx1eWW7 (0)
[7428:0605/130034.475:INFO:CONSOLE(3697)] "Uncaught (in promise) TypeError: Network request failed", source: serve://boundary/assets/vendor-c45cd3fe4fc053d713149ecaffed61a4.js (3697)
[7428:0605/130034.480:INFO:CONSOLE(4799)] "Uncaught (in promise) Error: Attempted to handle event `notFound` on <scope:global> while in state root.loaded.created.uncommitted. ", source: serve://boundary/assets/vendor-c45cd3fe4fc053d713149ecaffed61a4.js (4799)
[7428:0605/130034.506:INFO:CONSOLE(0)] "Access to XMLHttpRequest at 'https://boundary.abfelbaum.dev/v1/scopes?filter=(%22%2Fitem%2Ftype%22%20%3D%3D%20%22project%22)&recursive=true&scope_id=global' from origin 'serve://boundary' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource.", source: serve://boundary/#/scopes/global/authenticate/ampw_pducx1eWW7 (0)
[7428:0605/130034.509:INFO:CONSOLE(1158)] "Error while processing route: scopes.scope.projects.targets.index Network request failed TypeError: Network request failed
    at serve://boundary/assets/vendor-c45cd3fe4fc053d713149ecaffed61a4.js:3697:90", source: serve://boundary/assets/vendor-c45cd3fe4fc053d713149ecaffed61a4.js (1158)
```

Because of these errors I also modified the following things (in an attempt to fix the issue) with traefik wich runs in  front of boundary:

* Set `access-control-allow-origin: *` http header

I also reviewed https://github.com/hashicorp/boundary-ui/issues/1687, but I think that is another issue since I can get to the login interface.

**Comment by randallmorey** [comment]
Hi there. Thanks for trying Boundary and Boundary Desktop.  As you noted, CORS must be enabled for Boundary Desktop to connect to a controller.  There are a number of CORS configuration options to review in [TCP Listeners](https://developer.hashicorp.com/boundary/docs/configuration/listener/tcp#cors_enabled).  Start there and let us know if there's anything else we can help with.

**Comment by Abfelbaum** [comment]
Hey! Thanks for the fast answer!

I tried to enable cours like that:

![image](https://github.com/hashicorp/boundary-ui/assets/26802234/ea60e8b8-9708-4b7d-a3bb-43351ea52b18)

but that did not help with the issue. I also noticed that the `access-control-allow-origin` header was not set by boundary (or swallowed by traefik) and therefore I manually set the header in traefik - with no result.

Is there maybe a misconfiguration that i do not notice?

**Comment by randallmorey** [comment]
Thank you for the follow up. It appears as though the controller is now properly configured to allow incoming API requests from Boundary Desktop. Using [Boundary Desktop 1.5.1](https://releases.hashicorp.com/boundary-desktop/1.5.1/) I am able to connect to your controller and browse available auth methods.

<img width="1392" alt="Screenshot 2023-06-05 at 12 17 45" src="https://github.com/hashicorp/boundary-ui/assets/8390120/5ad6d9cd-36d3-4ebc-a179-f6ad6f30fe05">

**Comment by Abfelbaum** [comment]
Thank you for t he follow up too! :P 

As I stated in the issue description the problem is not getting to the sign in page, that was always possible. The error occures after logging in.

**Comment by randallmorey** [comment]
Which version of Boundary Desktop are you running?

**Comment by Abfelbaum** [comment]
I'm running Boundary Desktop version 1.5.1 (should be the latest?) on Windows 10.

**Comment by randallmorey** [comment]
It sounds like you're able to list scopes and auth methods in Desktop, and perform an initial login?  What errors, if any, do you see in logs after logging in?

I can't speak to Traefik, however they provide some [documentation on CORS configuration](https://doc.traefik.io/traefik/middlewares/http/headers/#cors-headers).

**Comment by guygrigsby** [comment]
Hello @Abfelbaum! We use Traefik to front our Boundary clusters as well. We let Traefik handle all the CORS headers in the same way their [documentation describes](https://doc.traefik.io/traefik/middlewares/http/headers/#cors-headers). That way Boundary doesn't have to worry about it. For example you could use a Traefik config that contains something similar to this:
```toml
[http.middlewares]
  [http.middlewares.cors.headers]
    accessControlAllowMethods = ["OPTIONS", "GET", "PATCH", "POST", "PUT", "DELETE", "HEAD"]
    accessControlAllowHeaders= ["Authorization", "Content-Type"]
    accessControlAllowOriginList = ["https://boundary.abfelbaum.dev"]
    accessControlMaxAge = 3600
```
and then also set the Boundary controller api listener block to include `cors_enabled = false`. Something like this:
```hcl
listener "tcp" {
	address = "10.1.0.2:9200"
	purpose = "api"
	tls_disable = true
	cors_enabled = false
}
```
The only thing I am not certain about here is the origin. If auth callbacks go through Traefik, it's possible that you'll need to add those explicitly also.
